### PR TITLE
Fixes: #1133 Add yyyy-MM-ddTHH:mm:ssX timezone aware format to defaul…

### LINKED ISSUE #1133: Add "yyyy-MM-dd'T'HH:mm:ssX" timezone aware format to default ones.
#### Summary

Add "yyyy-MM-dd'T'HH:mm:ssX" to the default date formats to support TZ comparison.

#### Example

```java
Calendar cal = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
cal.clear();
cal.set(2017, Calendar.NOVEMBER, 27, 20, 13, 24);
Date date = cal.getTime();

assertThat(date)
    .isEqualTo("2017-11-27T20:13:24+00:00");
```

See #1123

### PR DESCRIPTION
**Pull request by pswiatowski**
…t ones.

#### Check List:
* Fixes #1133 
* Unit tests : YES 
* Javadoc with a code example (API only) : NA

### CODE REVIEW

File: src/test/java/org/assertj/core/api/date/DateAssert_with_string_based_date_representation_Test.java [hunk 1]
```diff
@@ -73,12 +75,22 @@ public void date_assertion_should_support_timestamp_string_representation() thro
...
 
+  @Test
+  public void date_assertion_should_support_date_with_utc_time_zone_string_representation() throws ParseException {
+    SimpleDateFormat isoFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
+    isoFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+    Date date = isoFormat.parse("2003-04-26T00:00:00");
+
+    assertThat(date).isEqualTo("2003-04-26T00:00:00+00:00");
+  }
+
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/date/DateAssert_with_string_based_date_representation_Test.java]: can we add a test with the same point in time expressed in different timezones?

**Comment by joel-costigliola** [comment]
@pswiatowski thanks for the PR, looks good, just one test to add.

File: src/test/java/org/assertj/core/api/date/DateAssert_with_string_based_date_representation_Test.java (hunk 1 above)
Comment (pswiatowski) [reply src/test/java/org/assertj/core/api/date/DateAssert_with_string_based_date_representation_Test.java]: Done

File: src/test/java/org/assertj/core/api/date/DateAssert_with_string_based_date_representation_Test.java [hunk 2]
```diff
@@ -84,6 +84,21 @@ public void date_assertion_should_support_date_with_utc_time_zone_string_represe
...
+    SimpleDateFormat isoDateFormatUtc = new SimpleDateFormat(dateFormatPattern);
+    isoDateFormatUtc.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+    SimpleDateFormat isoDateFormatMadrid = new SimpleDateFormat(dateFormatPattern);
+    isoDateFormatMadrid.setTimeZone(TimeZone.getTimeZone("Europe/Madrid"));
+
+    Date date = isoDateFormatUtc.parse("2003-04-26T00:00:00");
+    String madridDate = isoDateFormatMadrid.format(date);
+
+    assertThat(date).isEqualTo(madridDate);
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/date/DateAssert_with_string_based_date_representation_Test.java]: this fails in travis ci:
```
DateAssert_with_string_based_date_representation_Test.date_assertion_should_support_date_with_different_utc_time_zone_string_representation:99 
expected:<2003-04-26T0[2]:00:00.000> but was:<2003-04-26T0[0]:00:00.000>
```
could it be that the picked date format is not the one supporting TZ ?

Comment (pswiatowski) [reply src/test/java/org/assertj/core/api/date/DateAssert_with_string_based_date_representation_Test.java]: Exactly. I replaced with Europe/Berlin.

Comment (joel-costigliola) [reply src/test/java/org/assertj/core/api/date/DateAssert_with_string_based_date_representation_Test.java]: I guess this is not a `madridDate` anymore ;-)

Comment (joel-costigliola) [reply src/test/java/org/assertj/core/api/date/DateAssert_with_string_based_date_representation_Test.java]: can we rename `madrid` stuff to `berlin` stuff is what I really wanted to say.

Comment (pswiatowski) [reply src/test/java/org/assertj/core/api/date/DateAssert_with_string_based_date_representation_Test.java]: Yes, but it failed with Berlin TZ in travis too.

Comment (pswiatowski) [reply src/test/java/org/assertj/core/api/date/DateAssert_with_string_based_date_representation_Test.java]: Yes, I will rename it.

**Comment by joel-costigliola** [comment]
Integrated thanks @pswiatowski !

**Pull request merged**